File: 4.ROS/Tasks/Task_4/nomeer_ws/src/count_pkg/scripts/pub_count.py
#!/usr/bin/python3

'''
count from 0 until user input ctrl+c
we need to publish count msg
'''

import rospy
from std_msgs.msg import Int32


def pub_count():
    # create publisher  node
    rospy.init_node("pub_count")

    # create publisher object & topic
    pub = rospy.Publisher("count_topic", Int32, queue_size=1) 
 
    rate = rospy.Rate(1) # publish count with 1 hz rate

    # define count msg
    count = 0

    # publish msg
    while not rospy.is_shutdown():
        rospy.loginfo(count)
        pub.publish(count)
        rate.sleep()  # ensure that node publish with 1 hz
        count +=1

if __name__ == "__main__":
    try:
        pub_count()
    except rospy.ROSInternalException:
        pass

File: Task_4/nomeer_ws/src/count_pkg/src/pub_count.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

int main(int argc, char** argv)
{
    ros::init(argc, argv,"pub_count_cpp");
    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<std_msgs::Int32>("count_topic_cpp",1);
    ros::Rate rate(1);

    std_msgs::Int32 count;
    count.data = 0;

    while (ros::ok())
    {
        pub.publish(count);
        ros::spinOnce();
        rate.sleep();
        ++count.data;
    }

    return 0;
}